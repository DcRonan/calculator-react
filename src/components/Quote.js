import Navbar from './Navbar';

const Quote = () => (
  <>
    <Navbar />
    <section className="quote">
      <h4 style={{ fontWeight: '500', fontSize: '30px' }}>
        Mathematics is not about numbers, equations, computations, or
        algorithms: it is about understanding. â€“William Paul Thurston
      </h4>
    </section>
  </>
);

export default Quote;
