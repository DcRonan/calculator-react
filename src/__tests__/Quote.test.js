import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Quote from '../components/Quote';

describe('checks if Quote component renders correctly', () => {
  test('should renders correct text', () => {
    render(
      <BrowserRouter>
        <Quote />
      </BrowserRouter>,
    );
    const text = screen.getByText(/Mathematics is not about numbers/i);
    expect(text).toBeInTheDocument();
  });

  test('should renders correct text', () => {
    render(
      <BrowserRouter>
        <Quote />
      </BrowserRouter>,
    );
    const text = screen.getByText('Quote');
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Quote />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
