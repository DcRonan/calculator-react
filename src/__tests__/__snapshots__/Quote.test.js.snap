// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Quote component renders correctly should render correctly 1`] = `
Array [
  <nav>
    <div>
      <div
        className="title"
      >
         Math Magicians 
      </div>
    </div>
    <div
      className="navbar"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </nav>,
  <section
    className="quote"
  >
    <h4
      style={
        Object {
          "fontSize": "30px",
          "fontWeight": "500",
        }
      }
    >
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston
    </h4>
  </section>,
]
`;
